azure-monitor-opentelemetry-distro/azure/monitor/opentelemetry/distro/diagnostic_logger.py

```diff
@@ -0,0 +1,87 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import logging
+from os import makedirs
+from os.path import exists, join
+
+from azure.monitor.opentelemetry.distro.constants import (
+    _CUSTOMER_IKEY,
+    _DIAGNOSTIC_LOG_PATH,
+    _EXTENSION_VERSION,
+    _IS_ON_APP_SERVICE,
+    _env_var_or_default,
+)
+from azure.monitor.opentelemetry.distro.version import VERSION
+
+_OPENTELEMETRY_DIAGNOSTIC_LOGGER_NAME = "opentelemetry"
+_DISTRO_DIAGNOSTIC_LOGGER_NAME = "azure.monitor.opentelemetry.distro"
+_DIAGNOSTIC_LOGGER_FILE_NAME = "applicationinsights-extension.log"
+_OPERATION = "Startup"
+_SITE_NAME = _env_var_or_default("WEBSITE_SITE_NAME")
+_SUBSCRIPTION_ID = _env_var_or_default("WEBSITE_OWNER_NAME")
+_opentelemetry_logger = logging.getLogger(
+    _OPENTELEMETRY_DIAGNOSTIC_LOGGER_NAME
+)
+_distro_logger = logging.getLogger(_DISTRO_DIAGNOSTIC_LOGGER_NAME)
+
+
+class AzureDiagnosticLogger:
+    _initialized = False
+
+    def initialize():
+        if not AzureDiagnosticLogger._initialized:
+            if _IS_ON_APP_SERVICE:
+                format = (
+                    "{"
+                    + '"time":"%(asctime)s.%(msecs)03d", '
+                    + '"level":"%(levelname)s", '
+                    + '"logger":"%(name)s", '
+                    + '"message":"%(message)s", '
+                    + '"properties":{'
+                    + f'"operation":"{_OPERATION}", '
+                    + f'"sitename":"{_SITE_NAME}", '
+                    + f'"ikey":"{_CUSTOMER_IKEY}", '
+                    + f'"extensionVersion":"{_EXTENSION_VERSION}", '
+                    + f'"sdkVersion":"{VERSION}", '
+                    + f'"subscriptionId":"{_SUBSCRIPTION_ID}", '
+                    + '"language":"python"'
+                    + "}"
+                    + "}"
+                )
+                if not exists(_DIAGNOSTIC_LOG_PATH):
+                    makedirs(_DIAGNOSTIC_LOG_PATH)
+                f_handler = logging.FileHandler(
+                    join(_DIAGNOSTIC_LOG_PATH, _DIAGNOSTIC_LOGGER_FILE_NAME)
+                )
+                formatter = logging.Formatter(
+                    fmt=format, datefmt="%Y-%m-%dT%H:%M:%S"
+                )
+                f_handler.setFormatter(formatter)
+                _opentelemetry_logger.addHandler(f_handler)
```

Comment:
We cannot be modifying the format of the `opentelemetry` logger. This logger is being shared across all opentelemetry modules.

Reply:
We are only changing the format of our handler and _adding_ that handler to their logger's level. It doesn't affect the other handlers that may be attached to their logger.